Developers/Unity Project/Spooksville/Assets/Scripts/Items/WeaponManager.cs

```diff
@@ -21,13 +16,57 @@ void Start()
         }
 
         RegisterWeapons();
+        Inventory.AddWeapon(GetWeaponByName("Candy Cane"));
+        Inventory.AddWeapon(GetWeaponByName("Driller"));
+        Inventory.AddWeapon(GetWeaponByName("A"));
+        Inventory.AddWeapon(GetWeaponByName("B"));
+        Inventory.AddWeapon(GetWeaponByName("C"));
+        Inventory.AddWeapon(GetWeaponByName("D"));
+        Inventory.AddWeapon(GetWeaponByName("E"));
+        Inventory.AddWeapon(GetWeaponByName("F"));
+        Inventory.AddWeapon(GetWeaponByName("G"));
+        Inventory.AddWeapon(GetWeaponByName("H"));
+    }
 
+    public void RegisterWeapon(Weapon weapon)
+    {
+        weapons.Add(weapon);
+    }
+
+    public List<Weapon> GetWeapons()
+    {
+        return weapons;
+    }
+
+    public Weapon GetRandomWeapon()
+    {
+        return weapons[(new System.Random()).Next(weapons.Count)];
+    }
+
+    public Weapon GetWeaponByName(string name)
```

Comment:
This is the exact kind of code you don't have to write if you use linq :)
Try
```
using System.Linq;
// ...
weapons.Select(x => x.Name.Equals(name)).Single();
```

Alternatively, if you are finding that you are primarily accessing an object in a collection by one property such as name, a `Dictionary<K, V>` might be more appropriate. However last time I checked they aren't editable in the editor so an array might be better if you want to do that.